com.unity.visualeffectgraph/Editor/Models/Contexts/VFXContext.cs

```diff
@@ -193,9 +194,54 @@ protected override void OnInvalidate(VFXModel model, InvalidationCause cause)
             }
         }
 
-        public virtual bool SetupCompilation() { return true; }
-        public virtual void EndCompilation() { }
+        private List<(string error, VFXErrorType type, string description)> m_CompilationErrors;
 
+        public virtual bool SetupCompilation()
+        {
+            m_CompilationErrors?.Clear();
+            return true;
+        }
+
+        public void RegisterCompilationError(string error, VFXErrorType type, string description)
+        {
+            m_CompilationErrors ??= new List<(string error, VFXErrorType type, string description)>();
+            m_CompilationErrors.Add((error, type, description));
+        }
+
+        public virtual void EndCompilation()
+        {
+            if (m_CompilationErrors != null && m_CompilationErrors.Any())
+            {
+                var errorLog = new StringBuilder();
+                var assetPath = AssetDatabase.GetAssetPath(this);
+                bool critical = false;
+                errorLog.Append($"The compilation of {assetPath} has generated errors");
+                errorLog.AppendLine();
+                foreach (var (_, type, description) in m_CompilationErrors)
+                {
+                    critical = critical || type == VFXErrorType.Error;
+                    errorLog.Append(description);
+                    errorLog.AppendLine();
+                }
+
+                if (critical)
+                    Debug.LogError(errorLog.ToString());
+                else
+                    Debug.LogWarning(errorLog.ToString());
```

Comment:
That's not a big deal, but here you are merging all messages in one single category (error or warning). Then, as soon as there's one error, warning messages will be logged as error, which can be misleading for the user, don't you think?

Reply:
These logs can occurs during compilation, I choose to aggregate error like this to make them more readable, however, you are right, it can be confusing to differentiate errors from warnings. I will add and `[Error]` and `[Warning]` in the log.

Reply:
Reolved at f12cb5e067269f7f26348b333ad983ae15065a51